### ISSUE TITLE
SinglebandGeoTiff construction

### ISSUE THREAD

**Issue by djohnson729** [locationtech/geotrellis#1975]
I was expecting the output from the following two code blocks to be the same, however in the second one, each pixel is shifted a half pixel down and to the right. I downloaded a sample landsat geotiff from s3://landsat-pds/L8 to use as input.

```
val image1 = SinglebandGeoTiff(imagePath)
image1.write(outputPath)
```
After running the above code and looking at the input and output images in QGIS, all the pixels match up perfectly.

```
val image1 = SinglebandGeoTiff(imagePath)
val image2 = SinglebandGeoTiff(image1.tile, image1.extent, image1.crs)
image2.write(outputPath)
```
After running the above code and looking at the input and output images in QGIS, the output pixels are shifted down and to the right by a half pixel. I was expecting them to be the same. After some investigation, I found that the reason for the shift is because the input image tags indicate pixels as points, but the output tags do not because they are defaulted. So when I instead ran the following, the input and output matched.
```
val image1 = SinglebandGeoTiff(imagePath)
val image2 = new SinglebandGeoTiff(image1.tile, image1.extent, image1.crs, image1.tags, image1.options)
image2.write(outputPath)
```
I suggest adding another apply method to the SinglebandGeoTiff companion object like the following just to make life a little easier.
```
  def apply(
    tile: Tile,
    extent: Extent,
    crs: CRS,
    tags: Tags,
    options: GeoTiffOptions
  ): SinglebandGeoTiff =
    new SinglebandGeoTiff(tile, extent, crs, tags, options)
```
Also, the crop methods in the SinglebandGeoTiff class will probably result in the same half pixel shift problem that I saw and should probably be changed to use the new apply method suggested above.

**Comment by lossyrob** [comment]
This is a great API suggestion @djohnson729, thanks.

**Comment by jbouffard** [comment]
@lossyrob Does it make sense to create the suggested, new `apply` method? Since `SinglebandGeoTiff` is a case class, I'm not sure what that would do, as doing `val image2 = image1.copy()` would yield the same result.

**Comment by lossyrob** [comment]
Ah, good catch.

@djohnson729 because SinglebandGeoTiff is a case class, that apply method is already on the object created by the compiler. So 

```scala
val image1 = SinglebandGeoTiff(imagePath)
val image2 = SinglebandGeoTiff(image1.tile, image1.extent, image1.crs, image1.tags, image1.options)
image2.write(outputPath)
```

should already work.

**Comment by lossyrob** [comment]
Crop methods fixed in #1976, @djohnson729 let us know if this works for you. This will be released in 1.0.1, which is slated for end of February.
